### PR TITLE
Kotlin: Exclude constructs in serialization constructors from `java/evaluation-to-constant`

### PR DESCRIPTION
**Pull request by tamasvajk**

### CODE REVIEW

File: java/ql/lib/semmle/code/java/frameworks/kotlin/Serialization.qll [hunk 1]
```diff
@@ -0,0 +1,16 @@
+/**
+ * Provides classes and predicates for working with thi `kotlinx.serialization` plugin.
```
Comment (igfoo) [inline java/ql/lib/semmle/code/java/frameworks/kotlin/Serialization.qll:2]: `thi` -> `the`

File: java/ql/src/Likely Bugs/Arithmetic/ConstantExpAppearsNonConstant.ql [hunk 2]
```diff
@@ -10,6 +10,7 @@
  */
 
 import java
+private import semmle.code.java.frameworks.kotlin.Serialization
```
Comment (igfoo) [inline java/ql/src/Likely Bugs/Arithmetic/ConstantExpAppearsNonConstant.ql:13]: `private` is unnecessary in a `.ql`

**Review by igfoo — COMMENTED** [review]
Is it possible to add a test for this?

**Comment by tamasvajk** [comment]
> Is it possible to add a test for this?

I think I can through integration tests (and without a lot of extra work). Let me try...

**Comment by tamasvajk** [comment]
> Is it possible to add a test for this?

@igfoo I added an integration test with [a constant expression](https://github.com/github/codeql/pull/10881/files#diff-d25cb911da8bcdc948a4bc15a20ceca84365a914b4271dd94d67272cdf0eab90R490-R492) in a [generated serialization constructor](https://github.com/github/codeql/pull/10881/files#diff-d25cb911da8bcdc948a4bc15a20ceca84365a914b4271dd94d67272cdf0eab90R476).

Could you please double check me on the [qlpack](https://github.com/github/codeql/pull/10881/files#diff-78d20e7c0f51a4918f30726692dcd054e16334143670770c5ae64df5b836691fR1-R3) changes? (I don't know enough about qlpacks.

**Review by igfoo — APPROVED** [review]
LGTM

**Review by igfoo — APPROVED** [review]

**Pull request merged**